### DISCUSSION TITLE
Trying to understand latency

### DISCUSSION THREAD

**Discussion by MaxYaffe** [jamulussoftware/jamulus#2318] [Technical Support & Help]
A friend and I are trying out Jamulus for remote choir practice.  We both are using Windows computers with direct Ethernet connections to our routers and USB mic+headphones headsets. 

We picked a public server that has 15 mS ping time but the overall latency was around 50mS.  What accounts for the other 35mS?   

I was reading one discussion that addresses this with the solution, don't listen to your signal, just to the Jamulus server.  Unfortunately that is impossible when you're a singer.

Thanks for your help.

**Comment by dcorson-ticino-com** [comment]
The ping time 15ms is the time used for the Internet communication from your computer to the server and back.
The rest is the time used by your audio interface and it's USB connection and any buffering.
35ms audio interface delay seems to be quite a bit, but you don't say how your buffering is set.
To optimise the buffering, open the Settings window, Audio/Network Setup tab.  There you can see and set the buffer size and Jitter buffer.
These setting will depend on the quality of your Internet connection.  If you have a very good, jitter-free, connection you can set the Buffer Delay to 2.67 ms which is the lowest buffer delay.  Set your Jitter buffer to Auto and see how it settles down.  Some connections can even profit from enabling the Small Network Buffers, That uses some extra bandwidth to lower the delay. 
If you have sound quality problems you will need to increase the buffer delay.  
Note that the speed of the Internet connection is not what counts.  It is the regularity of the communication, low jitter.  Jamulus will work very well on a 1Meg connection if the jitter is low. 
Play around with different values until you have reliable sound quality and that will be your best buffer setting.

Note that even as a singer you will have to get used to listening to the signal from the server in order to sing together with the other musicians.  This is not easy at the beginning, you need to anticipate the entries and always push a little with the tempo.  
You know that it is possible to not hear another person in a discussion, to over hear them.  This is exactly the same as what you need to do with your own voice.  After a while this becomes natural and you will find that you can play/sing well even with pretty big latencies.  Remember that when you listen to the server you hear exactly what the others hear so if you correct your singing so it is right it will be right for all the others.  Without listening to the server you can never know.
I play viola, another instrument which you can not "not listen to" while playing.  After playing now for over a year I don't even notice latency of under 60ms and have played successfully at over 100ms.  It is sometimes very wierd when playing, but the proof is in the music of the group.  It works.
